src/lib/address_resolve/AddressResolve_DefaultImpl.cpp

```diff
@@ -305,6 +306,8 @@ void Resolver::OnOperationalNodeResolved(const Dnssd::ResolvedNodeData & nodeDat
             }
 #endif
             result.address.SetIPAddress(nodeData.resolutionData.ipAddress[i]);
+            result.address.SetTransportType((result.supportsTcpClient || result.supportsTcpServer) ? Transport::Type::kTcp
```

Comment:
This does not seem right.  Just because the other side supports TCP, doesn't mean we will in fact use TCP....

Reply:
that's correct, that PR is not intended to gaurantee the usage of TCP, @pidarped  has other PRs that would depend on that plumbing to use TCP transport

Reply:
Yes, agree with @bzbarsky-apple. I think there is no need to set the address type here. As long as we are collecting the info that the peer supports TcpClient/Server, that can be used subsequently during session establishment phase. I also see that OperationalSessionSetup::UpdateDeviceData() takes the peerAddress and MRPConfig. I think we need to pass the ResolveResult so that OperationalSessionSetup can capture the TCP support bits for the peer and take the informed decision during EstablishConnection() to set the TransportType.

Reply:
sounds fair, i modified the code to route the result object to EstablishConnection method where the decision is made to support large payload session with a specific client